### PR TITLE
Homework11: Marie Ambrazhei 

### PR DESCRIPTION
**Pull request by MarieAmbrazhei**

### CODE REVIEW

File: homework11/homework11_1.py [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+@validate_arguments
+def result_of_validate(a, b):
+    """Prints the passed arguments"""
+    print(a, b)
+
+
+try:
+    result_of_validate(2, 4)
+except ValueError as e:
+    print(e)
```
Comment (Stml89) [inline homework11/homework11_1.py:37]: Just 
```suggestion

assert result_of_validate(2, 4) == <expected value>
```

File: homework11/homework11_4.py [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+def cache(func):
+    """ Decorator to cache the results of a function.
+    When the function is called with the same arguments,
+    it returns the cached result, avoiding repeated computations.
+   """
+
+    def wrapper(n):
+        if cached_results.get(n) is not None:
+            return cached_results[n]
+        result = func(n)
```
Comment (Stml89) [inline homework11/homework11_4.py:31]: It is better to define variable inside

```suggestion
    def wrapper(n):
        cached_results = {}
        if cached_results.get(n) is not None:
            return cached_results[n]
        result = func(n)
```

File: homework11/homework11_2.py [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+def is_number(func):
+    """Checks if the result of the function is a number."""
+
+    def wrapper(*args):
+        result = func(*args)
+        if not isinstance(result, (int, float)):
+            print(f'{result} is not a number.')
+            return None
+        return func(*args)
```
Comment (Stml89) [inline homework11/homework11_2.py:20]: You do not need to call `func` 2 times

**Review by Stml89 — CHANGES_REQUESTED** [review]

File: homework11/homework11_4.py (hunk 2 above)
Comment (MarieAmbrazhei) [reply homework11/homework11_4.py:31]: Sorry, but its not clear, if I move cached_result inside wrapper , I have something like: {}
{}
{}
{}
But if I use cached_result as global variable I have: {}
{1: 1}
{1: 1, 0: 0, 2: 1}
{1: 1, 0: 0, 2: 1, 3: 2}.
Are you sure that is good idea to place it cached_result inside wrapper? If so, I will move it inside the wrapper. Sorry but I am a little confused with this task :(((

File: homework11/homework11_1.py (hunk 1 above)
Comment (MarieAmbrazhei) [reply homework11/homework11_1.py:37]: fixed

File: homework11/homework11_2.py (hunk 3 above)
Comment (MarieAmbrazhei) [reply homework11/homework11_2.py:20]: fixed

File: homework11/homework11_4.py (hunk 2 above)
Comment (Stml89) [reply homework11/homework11_4.py:31]: My bad, it should be like this:
```
def cache(func):
    cached_results = {}  # словарь для хранения закэшированных результатов

    def wrapper(*args):
        if args in cached_results:  # проверяем, есть ли результат в кэше
            return cached_results[args]
        result = func(*args)  # вызываем функцию и сохраняем результат
        cached_results[args] = result  # сохраняем результат в кэше
        return result

    return wrapper
```

**Review by Stml89 — APPROVED** [review]
Fix it and you are good

**Pull request merged**